reader-v2.go

```diff
@@ -43,7 +43,12 @@ var packageBufferPool = sync.Pool{
 func getBuffer() ([]byte, func()) {
 	p := packageBufferPool.Get().(*[]byte)
 	return *p, sync.OnceFunc(func() {
-		clear(*p) // Clear to avoid leaking data between callers
+		// Defense-in-depth: explicitly zero buffer to prevent key material leakage
+		// clear() is used first (compiler optimized), followed by explicit zeroing
+		clear(*p)
+		for i := range *p {
+			(*p)[i] = 0
+		}
```

Comment:
No idea why it thinks that as needed. If it want to do something useful it could do...
```suggestion
		toZero := *p
		clear(toZero[:cap(toZero)])
```